app/Commands/GenerateTrackFilesCommand.php

```diff
@@ -142,16 +150,31 @@ public function execute()
 
             try {
                 foreach ($trackFiles as $trackFile) {
-                    $this->dispatch(new EncodeTrackFile($trackFile, false, true, $this->autoPublish));
+                    // Don't re-encode master files when replacing tracks with an already-uploaded version
+                    if ($trackFile->is_master && !$this->isForUpload && $this->isReplacingTrack) {
+                        continue;
+                    }
+                    $this->dispatch(new EncodeTrackFile($trackFile, false, false, $this->isForUpload, $this->isReplacingTrack));
                 }
 
             } catch (InvalidEncodeOptionsException $e) {
-                $this->track->delete();
+                // Only delete the track if the track is not being replaced
+                if ($this->isReplacingTrack) {
+                    $this->track->version_upload_status = Track::STATUS_ERROR;
+                    $this->track->update();
+                } else {
+                    $this->track->delete();
+                }
                 return CommandResponse::fail(['track' => [$e->getMessage()]]);
             }
 
         } catch (\Exception $e) {
-            $this->track->delete();
+            if ($this->isReplacingTrack) {
+                $this->track->version_upload_status = Track::STATUS_ERROR;
+                $this->track->update();
+            } else {
+                $this->track->delete();
+            }
```

Comment:
This section appears to be duplicating the `InvalidEncodeOptionsException` handler. I suggest either abstracting the common code into a helper function or combining them into a single exception handler.